### PR TITLE
Use tangermeme backend for tomtom

### PR DESCRIPTION
**Pull request by avantikalal**
Replaced the TOMTOM executable from the MEME suite with the tomtom reimplementation in tangermeme v0.4. Tomtom no longer has to be installed in the environment to use `run_modisco`, which is a major source of user issues.

Specific changes:
1. removed MEME suite installation in `Dockerfile`:
2. Created a `run_tomtom` function in `interpret.motifs`. This takes a dictionary of motifs and a meme file, and compares the motifs to the reference motifs in the meme file.
3. Added a test for `run_tomtom` in `tests/test_interpret.py`.
4. Modified `run_modisco` to use `run_tomtom` for the tomtom part if a meme file is provided. This involved creating a new function _add_tomtom_to_modisco_report which is modified from modiscolite. 
5. Reran all the tutorials.

Fully addresses #55

### CODE REVIEW

File: src/grelu/interpret/score.py [hunk 1]
```diff
@@ -205,6 +205,98 @@ def get_attributions(
...
 
 
+def _add_tomtom_to_modisco_report(
+    modisco_dir: str,
+    tomtom_results: pd.DataFrame,
+    meme_file: str,
+    top_n_matches: int,
+) -> None:
+    """
+    Modified from https://github.com/jmschrei/tfmodisco-lite/blob/main/modiscolite/report.py#L245
```
Comment (suragnair) [inline src/grelu/interpret/score.py:215]: can you add the permalink: https://github.com/jmschrei/tfmodisco-lite/blob/3c6e38f/modiscolite/report.py#L245

File: tests/test_interpret.py [hunk 2]
```diff
@@ -157,3 +158,73 @@ def test_scan_sequences():
...
+            0.013090962390804095,
+            0.017318747323072148,
+            3.348979505934935e-07,
+        ],
+        rtol=1e-3,
+    )
+    assert np.allclose(
+        df["E-value"],
+        [
+            2.679182916187628e-06,
```
Comment (suragnair) [inline tests/test_interpret.py:210]: are these number coming from running MEME suite TomTom itself or Jacob's TomTom? If not, would it be possible to check to be sure?

File: src/grelu/interpret/score.py (hunk 1 above)
Comment (suragnair) [reply src/grelu/interpret/score.py:215]: also, would it make more sense to put this function and run_modisco in a separate modisco.py file? Does not fit into score.py thematically I feel.

File: src/grelu/interpret/motifs.py [hunk 3]
```diff
@@ -339,3 +339,48 @@ def compare_motifs(
...
+
+    Args:
+        motifs: A dictionary whose values are Position Probability Matrices
+                (PPMs) of shape (4, L).
+        meme_file: Path to a meme file containing reference motifs.
+
+    Returns:
+        df: Pandas dataframe containing all tomtom results.
+    """
+    from statsmodels.stats.multitest import fdrcorrection
```
Comment (suragnair) [inline src/grelu/interpret/motifs.py:357]: is the import taken care of in Dockerfile and setup?

File: tests/test_interpret.py (hunk 2 above)
Comment (avantikalal) [reply tests/test_interpret.py:210]: Yes, these numbers were checked with both, they're almost identical

File: src/grelu/interpret/motifs.py (hunk 3 above)
Comment (avantikalal) [reply src/grelu/interpret/motifs.py:357]: I will add it to make sure

File: src/grelu/interpret/score.py (hunk 1 above)
Comment (avantikalal) [reply src/grelu/interpret/score.py]: done

**Review by suragnair — APPROVED** [review]

**Pull request merged**